lfs/transfer_queue.go

```diff
@@ -278,6 +306,21 @@ func (q *TransferQueue) run() {
 	}
 }
 
+func (q *TransferQueue) canRetry(err error, id string) bool {
+	if !IsRetriableError(err) {
+		return false
+	}
+
+	defer q.retrylock.Unlock()
+	q.retrylock.Lock()
```

Comment:
I think this would be fine. I'm lukewarm on pushing that responsibility into the Transferrable, but I do like avoiding the mutex.